tsl/src/nodes/decompress_chunk/exec.c

```diff
@@ -499,7 +499,7 @@ decompress_chunk_begin(CustomScanState *node, EState *estate, int eflags)
 		if (IsA(constified, Const))
 		{
 			Const *c = castNode(Const, constified);
-			if (c->constisnull || !DatumGetBool(c))
+			if (c->constisnull || !DatumGetBool(c->constvalue))
```

Comment:
This looks like it would have been broken before the fix, not only for PG16?

Reply:
Yes, I think you're right and I should fix the commit message! I only noticed it when compiling with pg16 though because I only got an error there.

Reply:
If you manage to figure out what was wrong and add tests, that would be great.